cli/cmd/validate/validate.go

```diff
@@ -0,0 +1,385 @@
+				if err != nil && !errors.Is(err, cmdutil.ErrNoMatchingProject) {
+					ch.PrintfWarn("Warning: failed to pull environment credentials: %v\n", err)
+				}
+			}
+
+			if err := enforceRepoLimits(cmd.Context(), projectPath, ch); err != nil {
+				return err
+			}
+
+			parsedLogFormat, ok := local.ParseLogFormat(logFormat)
+			if !ok {
+				return fmt.Errorf("invalid log format %q", logFormat)
+			}
+
+			envVarsMap, err := parseVariables(envVars)
+			if err != nil {
+				return err
+			}
+
+			parseErrors, err := parseProject(cmd.Context(), projectPath, environment)
+			if err != nil {
+				return err
+			}
+
+			// If there are parse errors, output them and exit
+			if len(parseErrors) > 0 {
+				result := &ValidationResult{
+					Success: false,
+					Summary: ValidationSummary{
+						ParseErrors: len(parseErrors),
+					},
+					ParseErrors: parseErrors,
+				}
+				return outputResult(ch, result, outputFormat, outputFile)
+			}
+
+			ch.Interactive = false
+			app, err := local.NewApp(cmd.Context(), &local.AppOptions{
+				Ch:             ch,
+				Verbose:        verbose,
+				Debug:          debug,
+				Reset:          reset,
+				Environment:    environment,
+				ProjectPath:    projectPath,
+				LogFormat:      parsedLogFormat,
+				Variables:      envVarsMap,
+				LocalURL:       "",           // No UI, so no local URL
+				AllowedOrigins: []string{""}, // No UI, so no allowed origins
+				ServeUI:        false,
+			})
+			if err != nil {
+				return err
+			}
+			defer app.Close()
+
+			return reconcileAndReport(cmd.Context(), ch, app, reconcileTimeout, outputFormat, outputFile)
+		},
+	}
+
+	validateCmd.Flags().SortFlags = false
+	validateCmd.Flags().StringSliceVarP(&envVars, "env", "e", []string{}, "Set environment variables")
+	validateCmd.Flags().BoolVar(&reset, "reset", false, "Clear and re-ingest source data")
+	validateCmd.Flags().BoolVar(&verbose, "verbose", false, "Sets the log level to debug")
+	validateCmd.Flags().BoolVar(&debug, "debug", false, "Collect additional debug info")
+	validateCmd.Flags().StringVar(&logFormat, "log-format", "console", "Log format (options: \"console\", \"json\")")
+	validateCmd.Flags().DurationVar(&reconcileTimeout, "reconcile-timeout", 60*time.Second, "Timeout for reconciliation (e.g. 60s, 2m)")
+	validateCmd.Flags().StringVar(&outputFormat, "output-format", "table", "Output format (options: \"table\", \"json\")")
+	validateCmd.Flags().StringVarP(&outputFile, "output", "o", "", "Output file for validation results (JSON format)")
+
+	return validateCmd
+}
+
+func resolveProjectPath(args []string) (string, error) {
+	if len(args) == 0 {
+		return ".", nil
+	}
+
+	projectPath := args[0]
+	if strings.HasSuffix(projectPath, ".git") {
+		repoName, err := gitutil.CloneRepo(projectPath)
+		if err != nil {
+			return "", fmt.Errorf("clone repo error: %w", err)
+		}
+		projectPath = repoName
+	}
+
+	return projectPath, nil
+}
```

Comment:
Instead of duplicating the logic from `rill start`, do you see a way we could unify more of this logic in `local.NewApp`?

Reply:
This does minimal needed things as compared to `rill start` like it does not check if current directory is as same user's home dir, also does not check for WSL partition. Thats why added it here directly and thus does not seem like duplication.

Reply:
I'm more thinking about future changes in `NewApp` now needing to be duplicated in `rill validate` (which people may miss, so the behavior goes out of sync). Stuff like home dir checks or WSL checks are pretty cheap, so would it matter if they also ran as part of `rill validate`?

It's okay to have duplication if you think it's much simpler or the logic will be diverge going forward, just wondered if that was the case.